### PR TITLE
bench: Represent paths with fs::path instead of std::string

### PR DESCRIPTION
**Pull request by ryanofsky**
Suggested https://github.com/bitcoin/bitcoin/pull/20744#issuecomment-1022486215

### CODE REVIEW

**Comment by MarcoFalke** [comment]
cr ACK 3278e64a22150e6941818deebd1622b715c0f0be

File: src/bench/bench.cpp [hunk 1]
```diff
@@ -24,13 +24,13 @@ const std::function<std::vector<const char*>()> G_TEST_COMMAND_LINE_ARGUMENTS{};
 
 namespace {
 
-void GenerateTemplateResults(const std::vector<ankerl::nanobench::Result>& benchmarkResults, const std::string& filename, const char* tpl)
+void GenerateTemplateResults(const std::vector<ankerl::nanobench::Result>& benchmarkResults, const fs::path& filename, const char* tpl)
```
Comment (hebasto) [inline src/bench/bench.cpp:27]: nit: s/filename/file/ ?

**Review by hebasto — COMMENTED** [review]
Concept ACK.

After changing type single quotes are redundant as `path` object is already quoted with double quotes:
https://github.com/bitcoin/bitcoin/blob/3ace3a17c9bce606cea05192f0da3ac62ac69dda/src/bench/bench.cpp#L37
https://github.com/bitcoin/bitcoin/blob/3ace3a17c9bce606cea05192f0da3ac62ac69dda/src/bench/bench.cpp#L40

---

While here, is it a good chance for some fixes?
1. It seems a negation should be used in the following message:
https://github.com/bitcoin/bitcoin/blob/3ace3a17c9bce606cea05192f0da3ac62ac69dda/src/bench/bench.cpp#L37
2. Shouldn't the following line https://github.com/bitcoin/bitcoin/blob/3ace3a17c9bce606cea05192f0da3ac62ac69dda/src/bench/bench.cpp#L40 be moved into the `if` branch above?

**Review by fanquake — APPROVED** [review]
ACK 3278e64a22150e6941818deebd1622b715c0f0be

File: src/bench/bench.cpp (hunk 1 above)
Comment (ryanofsky) [reply src/bench/bench.cpp]: > nit: s/filename/file/ ?

Sure, changed

**Review by ryanofsky — COMMENTED** [review]
Updated 3278e64a22150e6941818deebd1622b715c0f0be -> 824e1ffa9fd957d05e34f36abe381c2465d89702 ([`pr/bp.1`](https://github.com/ryanofsky/bitcoin/commits/pr/bp.1) -> [`pr/bp.2`](https://github.com/ryanofsky/bitcoin/commits/pr/bp.2), [compare](https://github.com/ryanofsky/bitcoin/compare/pr/bp.1..pr/bp.2)) with suggested changes. Thanks!

**Review by fanquake — APPROVED** [review]
untested ACK 824e1ffa9fd957d05e34f36abe381c2465d89702

**Review by hebasto — APPROVED** [review]
ACK 824e1ffa9fd957d05e34f36abe381c2465d89702, tested on Linux Mint 20.2 (x86_64).

**Pull request merged**